src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -195,7 +197,7 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
                 var exceptions = new List<Exception>();
 
                 // Execute the hosting startup assemblies
-                foreach (var assemblyName in _options.HostingStartupAssemblies)
+                foreach (var assemblyName in _options.HostingStartupAssemblies.Distinct(StringComparer.OrdinalIgnoreCase))
```

Comment:
I think it's worth logging a warning. We shouldn't silently ignore when a startup assembly is listed twice since that is likely wrong.